Samples/BackgroundBlur/DX11VideoRenderer/Common.h

```diff
@@ -0,0 +1,384 @@
+#pragma once
+
+#include <windows.h>
+#include <float.h> // for FLT_MAX
+#include <tchar.h>
+#include <math.h>
+#include <strsafe.h>
+#include <mmsystem.h>
+#include <mfapi.h>
+#include <mfidl.h>
+#include <evr.h>
+#include <mferror.h>
+#include <d3d11.h>
+#include <dxgi1_2.h>
+#include <dcomp.h>
+#include <wmcodecdsp.h> // for MEDIASUBTYPE_V216
+#include "DX11VideoRenderer.h"
+#include "linklist.h"
+#include "staticasynccallback.h"
+
+//DEFINE_GUID(CLSID_VideoProcessorMFT, 0x88753b26, 0x5b24, 0x49bd, 0xb2, 0xe7, 0xc, 0x44, 0x5c, 0x78, 0xc9, 0x82);
+
+// MF_XVP_PLAYBACK_MODE
+// Data type: UINT32 (treat as BOOL)
+// If this attribute is TRUE, the XVP will run in playback mode where it will:
+//      1) Allow caller to allocate D3D output samples
+//      2) Not perform FRC
+//      3) Allow last frame regeneration (repaint)
+// This attribute should be set on the transform's attrbiute store prior to setting the input type.
+DEFINE_GUID(MF_XVP_PLAYBACK_MODE, 0x3c5d293f, 0xad67, 0x4e29, 0xaf, 0x12, 0xcf, 0x3e, 0x23, 0x8a, 0xcc, 0xe9);
+
+namespace DX11VideoRenderer
+{
+    inline void SafeCloseHandle(HANDLE& h)
+    {
+        if (h != NULL)
+        {
+            CloseHandle(h);
+            h = NULL;
+        }
+    }
+
+    template <class T> inline void SafeDelete(T*& pT)
+    {
+        delete pT;
+        pT = NULL;
+    }
+
+    template <class T> inline void SafeDeleteArray(T*& pT)
+    {
+        delete[] pT;
+        pT = NULL;
+    }
+
+    template <class T> inline void SafeRelease(T*& pT)
+    {
+        if (pT != NULL)
+        {
+            pT->Release();
+            pT = NULL;
+        }
+    }
+
+    template <class T> inline double TicksToMilliseconds(const T& t)
+    {
+        return t / 10000.0;
+    }
+
+    template <class T> inline T MillisecondsToTicks(const T& t)
+    {
+        return t * 10000;
+    }
+
+    // returns the greatest common divisor of A and B
+    inline int gcd(int A, int B)
```

Comment:
There's a std::gcd template

Reply:
Oh this was copied from another repo... Feel free to leave it as-is if you don't wanna diverge from the original repo. Less conflicts will make it easier in the future if you want to pull new changes.

Reply:
Yes all of the files in common come from [mediafoundation samples](https://github.com/microsoft/Windows-classic-samples/tree/main/Samples/Win7Samples/multimedia/mediafoundation). I can at least not include the files I'm not using and maybe refactor their sample code later. All of the mf samples are from 5 years ago...